iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp

```diff
@@ -187,6 +187,13 @@ static bool isRootOp(Operation *op) {
       return true;
     }
   }
+
+  if (auto genericOp = dyn_cast<linalg::GenericOp>(op)) {
+    return llvm::all_of(genericOp.getIndexingMaps(), [](AffineMap map) {
+      return map.isProjectedPermutation();
+    });
+  }
+
```

Comment:
I feel that we will hit the same issue for indexed_generic op. If there is a gather op, it will be lowered to an indexed_generic op. After fusion, it is possible that an indexed_generic op to have "reduction" loop.